pkg/runtime/compstore/components.go

```diff
@@ -13,31 +13,73 @@ limitations under the License.
 
 package compstore
 
-import compsv1alpha1 "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+import (
+	"errors"
+	"fmt"
 
-func (c *ComponentStore) GetComponent(componentType, name string) (compsv1alpha1.Component, bool) {
+	compsv1alpha1 "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+)
+
+func (c *ComponentStore) GetComponent(name string) (compsv1alpha1.Component, bool) {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
 	for i, comp := range c.components {
-		if comp.Spec.Type == componentType && comp.ObjectMeta.Name == name {
+		if comp.ObjectMeta.Name == name {
 			return c.components[i], true
 		}
 	}
 	return compsv1alpha1.Component{}, false
 }
 
-func (c *ComponentStore) AddComponent(component compsv1alpha1.Component) {
+func (c *ComponentStore) AddPendingComponentForCommit(component compsv1alpha1.Component) error {
```

Comment:
Why need a pending queue as we add component one by one?

Reply:
I don't think it is necessarily true when we introduce hot reloading. Adding the name as pending means we can reserve name before committing when `Init` is successful.

Reply:
With hot reloading components can be updated in the sidecar in parallel, as opposed to being initialized sequentially when loaded at startup.